src/basins/basins_fractions.jl

```diff
@@ -0,0 +1,242 @@
+* clust_mod = "unsupervised" : clustering method chosen, can be "supervised" or "unsupervised". If "supervised", user needs to input the templates features also
+
+
+## Keywork arguments
+* T = 100 : total time for evolving initial conditions (after transient)
+* Ttr = 0 : transient time to evolve initial conditions
+* Δt = 1 : Integration time step, Δt = 1/fs, fs being the sampling frequency used in the bSTAB paper
+* y0 = []: initial conditions for the templates in the "supervised case"
+* templates_labels : (for supervised) labels for the templates
+* clust_params = NamedTuple() : other parameters for clustering method
+* diffeq = NamedTuple() : other parameters for the solvers of DiffEqs
+* extract_params = NamedTuplee() : other parameters for the feature_extraction function
+* min_neighbors = 10 : (for unsupervised) minimum number of neighbors (similar features) each feature needs to have in order to be considered in a cluster (fewer than this, it is labeled as an outlier, id=-1). This is somewhat hard to define, as it directly interferes with how many attractors the clustering finds. The authors use it equal to 10 always.
+
+## Available featurizers
+* lyapunov spectrum
+* statistical moments
+* whatever else in the original paper, which should be trivial to implement here
+
+
+## Description
+Let ``F(A)`` be the fraction the basin of attraction of an attractor ``A`` has in the
+chosen state space region ``\\mathcal{S}`` given by `sampler`. `basin_fractions` estimates ``F`` for all
+attractors in ``\\mathcal{S}`` by randomly sampling `N` initial conditions and counting which
+ones end up in which attractors. The error of this approach for each fraction is given by[^Menck2013]
+``e = \\sqrt{F(A)(1-F(A))}``.
+
+In `basin_fractions` we do not actually identify attractors, as e.g. done in [`basins_of_attraction`](@ref).
+Instead we follow the approach of Stender & Hoffmann[^Stender2021] which transforms each
+initial condition into a vector of features, and uses these to efficiently map initial
+conditions to attractors using a clustering technique.
+
+
+[^Menck2013]: Menck, Heitzig, Marwan & Kurths. How basin stability complements the linear stability paradigm. [Nature Physics, 9(2), 89–92](https://doi.org/10.1038/nphys2516)
+
+[^Stender2021]: Stender & Hoffmann, [bSTAB: an open-source software for computing the basin stability of multi-stable dynamical systems](https://doi.org/10.1007/s11071-021-06786-5)
+"""
+function basin_fractions(ds::DynamicalSystem, feature_extraction::Function, ic_grid::AbstractMatrix, clust_mod="supervised"; kwargs...)
+    #-- Integrate ICs to find features
+    # println("Integrating ICs to find features.")
+    feature_array = featurizer(ds, ic_grid,  feature_extraction; kwargs...)
+
+    #-- Classify features
+    # println("Classifying features in $(clust_mod) manner.")
+    if(clust_mod == "supervised")
```

Comment:
```suggestion
    if clust_mod == "supervised"
```

I have never-ever seen an `if` statement enclosing paranthesis. Where did you find that? Is it even a valid Julia syntax...?

Reply:
An old habit from C, I think. It is indeed valid, though unnecessary as you say. I will remove them.